types/set.go

```diff
@@ -294,3 +295,26 @@ func (s Set) IsNull() bool {
 func (s Set) IsUnknown() bool {
 	return s.Unknown
 }
+
+func (s Set) String() string {
+	if s.Unknown {
+		return attr.UnknownString
+	}
+
+	if s.Null {
+		return attr.NullString
+	}
+
+	var res strings.Builder
+
+	res.WriteString("(")
```

Comment:
Terraform also tends to bucket sets with the tuple/list "syntax" of square braces (e.g. `[`)